pkg/util/devices_test.go

```diff
@@ -142,6 +144,56 @@ var _ = Describe("SelectDynamicDirectPathIO", func() {
 	})
 })
 
+var _ = Describe("SelectVGPU", func() {
+	Context("selecting a vGPU device", func() {
+		It("will return only the selected device type", func() {
+			devOut := util.SelectVGPU(
+				[]vimTypes.BaseVirtualDevice{
+					newPCIPassthroughDevice(""),
+					&vimTypes.VirtualVmxnet3{},
+					newPCIPassthroughDevice("profile1"),
+					&vimTypes.VirtualSriovEthernetCard{},
+					newPCIPassthroughDevice(""),
+					newPCIPassthroughDevice("profile2"),
+				},
+			)
+			Expect(devOut).To(BeAssignableToTypeOf([]*vimTypes.VirtualPCIPassthrough{}))
+			Expect(devOut).To(HaveLen(2))
```

Comment:
Because of how newPCIPassthroughDevice() util function is working, It returns back a DDPIO device when the profile is "" and a vGPU when the profile is non-empty.